### PR TITLE
Pub/Sub Stream Binder code sample

### PR DESCRIPTION
**Pull request by joaoandremartins**
This is a code sample that sends messages using the Spring Cloud Stream
binder.

### CODE REVIEW

**Comment by joaoandremartins** [comment]
It would be great if anyone from Spring Cloud Stream could take a look here and let me know if there is anything else we should be conveying in this sample.
We don't really support consumer groups or partitioning FYI..

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/README.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+= Spring Cloud GCP Stream Binder for Pub/Sub Code Sample
+
+This code sample demonstrates how to use the Spring Cloud Stream binders for Google Cloud Pub/Sub.
```
Comment (artembilan) [inline spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/README.adoc]: s/binders/Binder ?

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/Message.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+package com.example;
+
+import java.time.LocalDateTime;
+
+/**
+ * @author João André Martins
+ */
+public class Message {
```
Comment (artembilan) [inline spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/Message.java]: Maybe call it `UserMessage` do not confuse with many existing `Message` classes ?

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+@EnableBinding(Source.class)
+@RestController
+public class SourceExample {
+
+	@Autowired
+	private Source source;
+
+	@PostMapping("/newMessage")
+	public Message sendMessage(@RequestParam("messageBody") String messageBody,
```
Comment (artembilan) [inline spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java]: So, why `@Publisher(Source.OUTPUT)` doesn't work here for you?

**Comment by sobychacko** [comment]
@joaoandremartins I just glanced through the code and it looks good to me. What happens if someone tries to set consumer group or partitoning properties? Does that throw an exception or no-op?

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java (hunk 3 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java]: Humm `@Publisher(Source.OUTPUT)` causes this method to return a 404. Do you have an idea why?

Comment (artembilan) [reply spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java]: Well, no idea, indeed.
Would you mind to share stack trace from logs on the matter?
Thanks

**Comment by joaoandremartins** [comment]
@sobychacko they're no-ops.
Setting a consumer group names adds the group suffix to the Pub/Sub subscription, but if I recall from previous conversations, there isn't actual consumer group distribution since Pub/Sub doesn't support that notion. i.e., a message sent to a topic X with subscriptions X.a and X.b is delivered to both subscriptions instead of just one.

**Comment by artembilan** [comment]
Checkstyle has failed for this build.
Also I still see a lot of compilation warning.

Didn't you fix them recently? What am I missing ?

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java (hunk 3 above)
Comment (artembilan) [reply spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java]: Right, there is a proxying.
So, let's never mind for now!
It's not important for this sample.

**Pull request merged**